### PR TITLE
Use numpy>=2

### LINKED ISSUE #598: NumPy 2.0.0 Breaking Changes
As of June 16th NumPy 2.0.0 has been released, updating a variety of functions and syntax. In many instances these changes are incompatible with the current state of the codebase and cause breaking changes. For instance, the change from `np.NaN` $\mapsto$ `np.nan` causes errors on line 82 of `modelinfo.py`:

```
def parse_parameter(name, units='', default=np.NaN,
                    user_limits=None, ptype='', description=''):
```

 Although previously installed instances of SasView will still be able to run using out-of-date versions of NumPy, this will cause errors for new installations of SasView.

### LINKED ISSUE #604: scipy 1.14 introduces breaking changes
As with `sasview` and `sasdata`, the `sasmodels` code base needs to be updated for the breaking changes in  `scipy 1.14`. Currently we are pinning to an older version of numpy and scipy for release 6.0 but this should not be forgotten.

### PR DESCRIPTION
**Pull request by krzywon**
This allows sasmodels to use numpy>=2. This also fixes two sphinx warnings that were happening during the build.

Fixes #604 
Fixes #598 
Refs #609

### CODE REVIEW

**Comment by bobleesj** [comment]
@krzywon Thanks for this PR, just wondering, will this PR be reviewed or merged?

**Comment by bobleesj** [comment]
@krzywon yes happy to review. Will do later in the afternoon.

File: .github/workflows/test.yml
Comment (bobleesj) [inline .github/workflows/test.yml:1]: I am wondering reasons preventing `sasmodels` from testing against Python 3.12 in the test.yml workflow?

File: setup.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def find_version(package):
                 return version[1:-1]
     raise RuntimeError("Could not read version from %s/__init__.py"%package)
 
-install_requires = ['numpy==1.*', 'scipy']
+install_requires = ['numpy', 'scipy']
```
Comment (bobleesj) [inline setup.py:32]: great

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -36,8 +36,7 @@ jobs:
         python -m pip install --upgrade pip
         python -m pip install wheel setuptools
         python -m pip install mako
-        python -m pip install numpy==1.*
```
Comment (bobleesj) [inline .github/workflows/test.yml:39]: great

**Review by bobleesj — COMMENTED** [review]
@krzywon reviewed. Could you please check my comments?

File: .github/workflows/test.yml
Comment (krzywon) [reply .github/workflows/test.yml:1]: Great question! I don't think there is anything holding us back. I'm going to add 3.12 and see what happens.

File: setup.py [hunk 3]
```diff
@@ -70,8 +70,8 @@ def find_version(package):
     },
     install_requires=install_requires,
     extras_require={
-        'full': ['docutils', 'bumps', 'matplotlib', 'columnize'],
-        'server': ['bumps'],
+        'full': ['docutils', 'bumps==0.*', 'matplotlib', 'columnize', 'siphash24'],
```
[the remark this answers is not in the document]
Comment (krzywon) [reply setup.py:73]: `siphash24` was optionally required for a set of dependencies, but these changes move to a point where it is now required, which is why I've added it here. Without it, our tests were failing.

`bumps` is our optimizing package (https://github.com/bumps/bumps). v1.0 of bumps has made significant, breaking, changes that will create issues with packages that depend on sasmodels. This is a temporary fix to maintain compatibility.

**Review by sbillinge — APPROVED** [review]
Thanks for the review @bobleesj .  This also looks good to me.

**Comment by krzywon** [comment]
@bobleesj, I've added the py3.13 test, as requested. Let's see what happens!

@sbillinge, I spoke to @pkienzle, and he says he will take a look when he has a chance.

**Comment by krzywon** [comment]
I rolled back from py3.13. The build is failing to find a compatible pyopencl wheel. I could probably build from source, but that is beyond the scope of this PR.

**Comment by bobleesj** [comment]
> I rolled back from py3.13. The build is failing to find a compatible pyopencl wheel. I could probably build from source, but that is beyond the scope of this PR.

Yes. Thanks a ton!

**Comment by andyfaff** [comment]
For reference, https://numpy.org/doc/stable/numpy_2_0_migration_guide.html

**Review by pkienzle — COMMENTED** [review]
Oops... thought I approved this already. Sorry!

**Review by pkienzle — APPROVED** [review]

**Pull request merged**